pkg/admission/admission.go

```diff
@@ -255,23 +255,30 @@ func (a *Admission) validateAlertManagerConfig(ar v1.AdmissionReview) *v1.Admiss
 	if ar.Request.Resource != alertManagerConfigGVR {
 		err := fmt.Errorf("expected resource to be %v, but received %v", alertManagerConfigResource, ar.Request.Resource)
 		level.Warn(a.logger).Log("err", err)
-		a.amConfValidationErrorsCounter.Inc()
+		a.incrementCounter(a.amConfValidationErrorsCounter)
 		return toAdmissionResponseFailure("Unexpected resource kind", alertManagerConfigResource, []error{err})
 	}
 
 	amConf := &monitoringv1alpha1.AlertmanagerConfig{}
 	if err := json.Unmarshal(ar.Request.Object.Raw, amConf); err != nil {
 		level.Info(a.logger).Log("msg", errUnmarshalConfig, "err", err)
-		a.amConfValidationErrorsCounter.Inc()
+		a.incrementCounter(a.amConfValidationErrorsCounter)
 		return toAdmissionResponseFailure(errUnmarshalConfig, alertManagerConfigResource, []error{err})
 	}
 
 	if err := alertmanager.ValidateConfig(amConf); err != nil {
 		msg := "invalid config"
 		level.Debug(a.logger).Log("msg", msg, "content", amConf.Spec)
 		level.Info(a.logger).Log("msg", msg, "err", err)
-		a.amConfValidationErrorsCounter.Inc()
+		a.incrementCounter(a.amConfValidationErrorsCounter)
 		return toAdmissionResponseFailure("AlertManagerConfig is invalid", alertManagerConfigResource, []error{err})
 	}
 	return &v1.AdmissionResponse{Allowed: true}
 }
+
+// todo - this can be removed when deprecated metrics are removed
```

Comment:
```suggestion
// TODO(PhilipGough) - this can be removed when following deprecated metrics are removed:
// - prometheus_operator_rule_validation_errors_total
//- prometheus_operator_alertmanager_config_validation_errors_total 
```

nit to make our future easier ;)